lib/RGBot.js

```
@@ -294,83 +593,153 @@ const RGBot = class {
+            return new FindResult(pt, sortValue)
+        }).sort((a,b) => {
+            return a.value - b.value
         })
+
+        result.length = Math.min(result.length, maxCount || result.length)
+
+        return result
     }
 
     /**
      * The Bot will approach the given Entity.
-     * @param {Entity | Item} entity The Entity to approach
-     * @param {object} options={} Optional parameters
+     * @param {Entity} entity The Entity to approach
+     * @param {object} [options={}] Optional parameters
      * @param {number} [options.reach=1] The Bot will approach and stand within this reach of the Entity
      * @returns {Promise<boolean>} true if the Bot successfully reaches the Entity, else false
      */
     async approachEntity(entity, options = {}) {
         const reach = options.reach || 1;
         if (!entity) {
-            console.error(`approachEntity: Entity was null or undefined`);
+            this.#logWarn(`approachEntity: Entity was null or undefined`);
             return false;
         } else {
             try {
-                const rangeToEntity = this.#bot.entity.position.distanceTo(entity.position);
-                this.#log(`Approaching ${this.getEntityName(entity)} from range ${rangeToEntity} within reach of ${reach}`);
+                const rangeToEntity = this.position().distanceTo(entity.position);
+                this.#logDebug(`approachEntity: Approaching ${this.getEntityName(entity)} from range ${rangeToEntity} within reach of ${reach}`);
                 const goal = new GoalNear(entity.position.x, entity.position.y, entity.position.z, reach);
                 return await this.handlePath(async () => {
                     await this.#bot.pathfinder.goto(goal);
                 });
             } catch(err) {
-                console.error(`approachEntity: failed to approach ${this.getEntityName(entity)}`)
+                this.#log(`approachEntity: failed to approach ${this.getEntityName(entity)}`)
```

We hide the error, which could make debugging difficult - maybe pass that along since the logging accepts exceptions?